### PR TITLE
Gracefully shutdown worker threads

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Gracefully shutdown worker threads when TCPReqServerChannel is closed or destroyed.

### Previous Behavior

Running thread causes processes not to terminate.

### New Behavior

Threads will stop gracefully and processes will terminate as they should. This fixes the windows python 3 Jenkins build.

### Tests written?

No - Fixes test suite

### Commits signed with GPG?

No

### CODE REVIEW

**Review by DmitryKuzmenko — APPROVED** [review]

**Review by gtmanfred — APPROVED** [review]

File: salt/transport/tcp.py [hunk 1]
```diff
@@ -556,6 +557,12 @@ def close(self):
                     raise exc
             self._socket.close()
             self._socket = None
+        if hasattr(self.req_server, 'stop'):
+            try:
+                self.req_server.stop()
+            except Exception as exc:
+                log.debug("TCPReqServerChannel close generated an exception: %s" str(exc))
```
Comment (rallytime) [inline salt/transport/tcp.py]: This has a syntax error: there is a missing comma after `%s"`.

File: salt/transport/tcp.py [hunk 2]
```diff
@@ -556,6 +557,11 @@ def close(self):
                     raise exc
             self._socket.close()
             self._socket = None
+        if hasattr(self.req_server, 'stop'):
+            try:
+                self.req_server.stop()
+            except Exception as exc:
+                log.debug("TCPReqServerChannel close generated an exception: %s", str(exc))
```
Comment (cachedout) [inline salt/transport/tcp.py]: I think `log.exception` would be better here.

**Review by cachedout — APPROVED** [review]
I'll approve this but with a comment that can be handled in a follow-up PR.

File: salt/transport/tcp.py (hunk 2 above)
Comment (dwoz) [reply salt/transport/tcp.py]: I think I can get rid of that completely by checking is_alive in the `stop` method. If not, I'll change this to log exception.

**Pull request merged**